### PR TITLE
Fix stroke Trim support in cases where strokeStart > strokeEnd

### PR DESCRIPTION
**Pull request by calda**
This PR fixes stroke `Trim` support in cases where `strokeStart` > `strokeEnd`.

`CAShapeLayer` requires that `strokeStart` be less than `strokeEnd`. If we detect that the value for `trim.start` is always greater than `trim.end`, we swap the values to match what `CAShapeLayer` expects.

| Before | After |
| ----- | ----- |
|  |  |

### CODE REVIEW

File: Sources/Private/Experimental/Animations/ShapeAnimation.swift [hunk 1]
```diff
@@ -77,9 +77,24 @@ extension CAShapeLayer {
...
+    //    we swap the start/end keyframes to match what `CAShapeLayer` expects.
+    let strokeStartKeyframes: KeyframeGroup<Vector1D>
+    let strokeEndKeyframes: KeyframeGroup<Vector1D>
+    if trim.startValueIsAlwaysGreaterThanEndValue() {
+      strokeStartKeyframes = trim.end
+      strokeEndKeyframes = trim.start
+    } else {
+      strokeStartKeyframes = trim.start
+      strokeEndKeyframes = trim.end
+    }
```
Comment (erichoracek) [inline Sources/Private/Experimental/Animations/ShapeAnimation.swift:93]: nit: might be simpler to reason about if this method returns a tuple of the start/end keyframes:
```suggestion
    let (strokeStartKeyframes, strokeEndKeyframes) = trim.caShapeLayerKeyframes()
```

File: Sources/Private/Experimental/Animations/ShapeAnimation.swift [hunk 2]
```diff
@@ -98,3 +113,28 @@ extension CAShapeLayer {
...
+  ///    `strokeStart` be less than `strokeEnd`.
+  fileprivate func startValueIsAlwaysGreaterThanEndValue() -> Bool {
+    let keyframeTimes = Set(start.keyframes.map { $0.time } + end.keyframes.map { $0.time })
+
+    let startInterpolator = KeyframeInterpolator(keyframes: start.keyframes)
+    let endInterpolator = KeyframeInterpolator(keyframes: end.keyframes)
+
+    for keyframeTime in keyframeTimes {
+      let startAtTime = startInterpolator.value(frame: keyframeTime) as! Vector1D
+      let endAtTime = endInterpolator.value(frame: keyframeTime) as! Vector1D
```
Comment (erichoracek) [inline Sources/Private/Experimental/Animations/ShapeAnimation.swift:131]: nit: include a comment on why this force cast is safe?

**Review by erichoracek — APPROVED** [review]
LGTM!

File: Sources/Private/Experimental/Animations/ShapeAnimation.swift (hunk 2 above)
Comment (calda) [reply Sources/Private/Experimental/Animations/ShapeAnimation.swift:131]: oh whoops I meant to clean this up, ty for the reminder

File: Sources/Private/Experimental/Animations/ShapeAnimation.swift (hunk 1 above)
Comment (calda) [reply Sources/Private/Experimental/Animations/ShapeAnimation.swift:93]: ooohh good idea

**Pull request merged**